packages/datadog-plugin-kafkajs/src/producer.js

```diff
@@ -20,6 +29,7 @@ class KafkajsProducerPlugin extends ProducerPlugin {
 
     for (const message of messages) {
       if (typeof message === 'object') {
+        if (this.config.dsmEnabled) message.headers['dd-pathway-ctx'] = pathwayCtx
```

Comment:
Similar to the above, this could be hidden away in a base class method.

Reply:
I think we'll move it once we start supporting more queue products next quarter